granulate_utils/metadata/databricks_client.py

```diff
@@ -132,15 +130,89 @@ def _cluster_all_tags_metadata(self) -> Optional[Dict[str, str]]:
             env = response.json()
         except Exception as e:
             raise DatabricksJobNameDiscoverException(f"Environment request failed {response.text=}") from e
-        props = env.get("sparkProperties")
-        if props is None:
-            raise DatabricksJobNameDiscoverException(f"sparkProperties was not found in {env=}")
-        for prop in props:
-            if prop[0] == CLUSTER_TAGS_KEY:
-                try:
-                    all_tags_value = json.loads(prop[1])
-                except Exception as e:
-                    raise DatabricksJobNameDiscoverException(f"Failed to parse {prop=}") from e
-                return {cluster_all_tag["key"]: cluster_all_tag["value"] for cluster_all_tag in all_tags_value}
+        return env
+
+    @staticmethod
+    def _extract_service_name_candidates(spark_properties: Any) -> Dict[str, str]:
+        # Creating a dict of the relevant properties and their values.
+        relevant_props = [CLUSTER_ALL_TAGS_PROP, CLUSTER_NAME_PROP]
+        service_name_prop_candidates = {prop[0]: prop[1] for prop in spark_properties if prop[0] in relevant_props}
+        if len(service_name_prop_candidates) == 0:
+            # We expect at least one of the properties to be present.
+            raise DatabricksJobNameDiscoverException(
+                f"Failed to create dict of relevant properties {spark_properties=}"
+            )
+        return service_name_prop_candidates
+
+    def _cluster_all_tags_metadata(self) -> Optional[Dict[str, str]]:
+        """
+        Returns `includes spark.databricks.clusterUsageTags.clusterAllTags` tags as `Dict`.
+        In any case this function returns `None`, a retry is required.
+        """
+        if not os.path.isfile(DATABRICKS_METRICS_PROP_PATH):
+            # We want to retry in case the cluster is still initializing, and the file is not yet deployed.
+            return None
+        # Discovering SparkUI apps url.
+        self._discover_apps_url()
+        if self._is_apps_url_discovered() is False:
```

Comment:
```suggestion
        if not self._is_apps_url_discovered():
```

Reply:
Or if you take my suggestion from the previous comments this can just be:
```
    if (apps_url := self._get_apps_url()) is None:
        # SparkUI apps url was not discovered, retrying.
        return None
```

Reply:
2ed7a37